Lib/tempfile.py

```diff
@@ -98,7 +98,7 @@ def _infer_return_type(*args):
     """Look at the type of all args and divine their implied return type."""
     return_type = None
     for arg in args:
-        if arg is None:
+        if arg is None or isinstance(arg, _os.PathLike):
```

Comment:
ah, I get it. Is my misunderstand about path-like objects, stdlib implement __fspath__ with return `str`. And there should be some 3rd-party class implement __fspath__ with return `bytes`.